sigstore/_cli.py

```diff
@@ -416,24 +465,31 @@ def _verify(args: argparse.Namespace) -> None:
         if not file.is_file():
             args._parser.error(f"Input must be a file: {file}")
 
-        sig, cert = args.signature, args.certificate
+        sig, cert, bundle = args.signature, args.certificate, args.rekor_bundle
         if sig is None:
             sig = file.parent / f"{file.name}.sig"
         if cert is None:
             cert = file.parent / f"{file.name}.crt"
+        if bundle is None:
+            bundle = file.parent / f"{file.name}.rekor"
 
         missing = []
         if not sig.is_file():
             missing.append(str(sig))
         if not cert.is_file():
             missing.append(str(cert))
+        if not bundle.is_file() and args.require_rekor_offline:
```

Comment:
Are you requiring that the cert and sig be passed in addition to the bundle? For example, with Cosign, if you give us a bundle, we'll extract the cert and signature automatically - https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/verify/verify_blob.go#L173

cc @asraa for more thoughts

Reply:
Yep -- right now the only "bundle" we support is the "Rekor only" bundle, i.e. the one in the [signature spec](https://github.com/sigstore/cosign/blob/main/specs/SIGNATURE_SPEC.md). We don't currently support the "cosign-style" bundle with embedded cert and signature (although we plan to support the "sigstore-style" one, once it's finalized).

TL;DR `sigstore-python` currently expects 2+ independent verification inputs for each `verify` invocation: a separate `.crt`, separate `.sig`, and optionally a separate `.rekor` containing the JSON encoded Rekor bundle.

Reply:
For Cosign, it's the "rekor bundle" where we extract the signature and certificate. You could argue it's putting additional trust in Rekor though.